composer/trainer/trainer.py

```diff
@@ -1814,8 +1814,10 @@ def _train_microbatch(self, use_grad_scaling: bool, current_batch_size: int, tot
                     loss.mul_(microbatch_num_samples / current_batch_size)
                     total_loss += loss.detach().clone()
             else:
+                final_loss = self._device.tensor_to_device(torch.zeros(size=(1,)))
```

Comment:
Do we need to worry about the dtype of loss? Or will torch convert the dtype if necessarry?

Reply:
I think it should be fine? I did this because `total_loss` is set this way, should I just do `final_loss = 0`?

Reply:
Just manually tested it, and torch keeps the more precise dtype, so this should be fine.